src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -137,18 +137,28 @@ int halide_hexagon_remote_power_hvx_off() {
     return 0;
 }
 
-int halide_hexagon_remote_set_performance(
+int halide_hexagon_remote_set_performance_v2(
```

Comment:
Sorry I didn't notice this before, but it looks like the only thing that is used anymore is `mode` and `latency`, unless it falls back to the old version?

Right now, the flow is a bit back-and-forth:

- set_performance_mode takes a mode, translates that into a bunch of bandwidth, mips, etc. settings.
- set_performance_mode then calls set_performance_v2 with those made up settings.
- set_performance_v2 proceeds to ignore most of those made up settings, and just look at the mode.
- ...unless the new setting isn't supported, then it falls back to set_performance using all of the made-up settings.

Instead, it seems like the flow should be:

- set_performance_mode calls `HAP_power_set` with `HAP_power_set_DCVS_v2`
- if that fails, *then* call set_performance (which now doesn't need to be modified to take a mode parameter anymore).

Reply:
In that case then, is there a need for shim_host changes?

Reply:
I don't think so, the simulator doesn't have any of the performance APIs (they are optional).